controllers/providers/aws/aws.go

```diff
@@ -151,15 +153,32 @@ func (w *AwsWorker) InstanceProfileExist(name string) (*iam.InstanceProfile, boo
 	return out.InstanceProfile, true
 }
 
-func (w *AwsWorker) GetBasicBlockDevice(name, volType string, volSize int64) *autoscaling.BlockDeviceMapping {
-	return &autoscaling.BlockDeviceMapping{
+func (w *AwsWorker) GetBasicBlockDevice(name, volType, snapshot string, volSize, iops int64, delete, encrypt *bool) *autoscaling.BlockDeviceMapping {
+	device := &autoscaling.BlockDeviceMapping{
 		DeviceName: aws.String(name),
 		Ebs: &autoscaling.Ebs{
-			VolumeSize:          aws.Int64(volSize),
-			VolumeType:          aws.String(volType),
-			DeleteOnTermination: aws.Bool(true),
+			VolumeType: aws.String(volType),
 		},
 	}
+	if delete != nil {
+		device.Ebs.DeleteOnTermination = delete
+	} else {
+		device.Ebs.DeleteOnTermination = aws.Bool(true)
+	}
+	if encrypt != nil {
+		device.Ebs.Encrypted = encrypt
+	}
+	if iops != 0 {
+		device.Ebs.Iops = aws.Int64(iops)
```

Comment:
can we add this validation https://github.com/aws/aws-sdk-go/blob/v1.33.16/service/autoscaling/api.go#L9756

Reply:
I can add a validation that IOPS is not 1-99

Reply:
added